src/physics/em/EPlusGGInteractor.i.hh

```diff
@@ -24,38 +31,96 @@ EPlusGGInteractor::EPlusGGInteractor(const EPlusGGInteractorPointers& shared,
 {
     REQUIRE(inc_energy_ >= this->min_incident_energy()
             && inc_energy_ <= this->max_incident_energy());
-    REQUIRE(particle.def_id() == shared_.gamma_id); // XXX
+    REQUIRE(particle.def_id() == shared_.positron_id);
 }
 
 //---------------------------------------------------------------------------//
 /*!
- * Sample using the XXX model.
+ * Sample two gammas using the G4eeToTwoGammaModel model.
  */
 template<class Engine>
 CELER_FUNCTION Interaction EPlusGGInteractor::operator()(Engine& rng)
 {
-    // Allocate space for XXX (electron, multiple particles, ...)
-    Secondary* secondaries = this->allocate_(0); // XXX
+    // Allocate space for two gammas
+    Secondary* secondaries = this->allocate_(2);
     if (secondaries == nullptr)
     {
-        // Failed to allocate space for a secondary
+        // Failed to allocate space for two secondaries
         return Interaction::from_failure();
     }
 
-    // XXX sample
-    (void)sizeof(rng);
-
-    // Construct interaction for change to primary (incident) particle
+    // Construct interaction for change to the incident positron
     Interaction result;
-    result.action      = Action::scattered;                     // XXX
-    result.energy      = units::MevEnergy{inc_energy_.value()}; // XXX
-    result.direction   = inc_direction_;
-    result.secondaries = {secondaries, 1}; // XXX
-
-    // Save outgoing secondary data
-    secondaries[0].def_id    = shared_.electron_id; // XXX
-    secondaries[0].energy    = units::MevEnergy{0}; // XXX
-    secondaries[0].direction = {0, 0, 0};           // XXX
+    result.action      = Action::absorbed;
+    result.energy      = units::MevEnergy{0}; 
+    result.direction   = {0, 0, 0};
```

Comment:
We'll need the same "interaction from absorption" code in other interaction kernels as well. Can you move this to a new function `Interaction::from_absorption()` (in the `Interaction.hh` file) so we can reuse it in the other interactors as well?

Reply:
Added Interaction::from_absorption() in Interaction.hh and replaced those lines with it.